geonotebook/wrappers/image.py

```diff
@@ -37,3 +45,48 @@ def bounds(self):
                     self.dataset.bounds.top,
                     self.dataset.bounds.right,
                     self.dataset.bounds.bottom)
+
+    def _get_band_tag(self, index, prop, convert=float):
+        assert not index < 0, \
+            IndexError("Bands are indexed from 1")
+
+        assert not index > self.count, \
+            IndexError("Band index out of range")
+
+        return convert(self.dataset.tags(index)[prop])
```

Comment:
This will crash if tags are not set. Compute this if the statistics are not available. This includes pulling NO_DATA field to ensure we do get the right statistics. There should be a public interface for getting NO_DATA value